src/licensetexts.ts

```diff
@@ -0,0 +1,123 @@
+/**
+This file is part of CycloneDX generator for NPM projects.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+import { Enums, Models } from '@cyclonedx/cyclonedx-library'
+import * as fs from 'fs'
+import { join } from 'path'
+
+import { PropertyNames } from './properties'
+
+/**
+ * Returns the local installation path of the component, which is mentioned in the component
+ *
+ * @param {Models.Component} component
+ * @returns {string} installation path
+ */
+function getComponentInstallPath (component: Models.Component): string {
+  for (const property of component.properties) {
+    if (property.name === PropertyNames.PackageInstallPath) {
+      return (property.value)
+    }
+  }
+  return ''
+}
+
+/**
+ * Searches typical files in the package path which have typical a license text inside
+ *
+ * @param {string} pkgPath
+ * @param {string} licenseName
+ * @returns {Map<string, string>} filepath as key and guessed content type as value
+ */
+function searchLicenseSources (pkgPath: string, licenseName: string): Map<string, string> {
+  const licenseFilenamesWType = new Map<string, string>()
+  if (pkgPath.length < 1) {
+    return licenseFilenamesWType
+  }
+  const typicalFilenames = ['LICENSE', 'License', 'license', 'LICENCE', 'Licence', 'licence', 'NOTICE', 'Notice', 'notice']
```

Comment:
Please change the list/implementation to be case-insensitive.
I'd duggest to have only lowercased entries in the list.

Reply:
I chose the other way first to get a list of all files and match case-insensitive. Maybe there is later an idea to handle multiple files